### DISCUSSION TITLE
Resource Roundup

### DISCUSSION THREAD

**Discussion by BekahHW** [Virtual-Coffee/virtualcoffee.io#496] [Technical Resources]
With an active and growing community like ours, we get a lot of folks asking for resources to learn different things. Let's start collecting resources we personally have used and approve of or know from a reliable source/sources that it's a good resource so we can create a VC Recommended Resource Guide.

If you can, please use this format:

## Tech Category - specific language
[Resource Title](link) - **type of resource and author** -  short description or main topic - **COST**: (if applicable)

For example:

## Frontend - React/Gatsby
[Accessibility in JavaScript Applications](https://frontendmasters.com/courses/javascript-accessibility/) - **Frontend Masters Course - Marcy Sutton** - We can produce innovative and inclusive JavaScript-powered web apps! This course will teach you to remove barriers to access that might prevent people with disabilities from using a modern JavaScript web application. We’ll study accessibility in UI components, primarily with React and Gatsby.js, but with knowledge applicable to all JavaScript-heavy web stacks. Debug your sites for accessibility, manage keyboard focus, announce live updates to the page to screen readers, and use manual and automated testing to gain web accessibility superpowers! - **COST**: Frontend Masters Subscription at $39/month with options for yearly and team.

**Comment by glenmccallumcan** [comment]
## Functional Programming - Haskell
[Haskell Programming from First Principles](https://haskellbook.com/) - Christopher Allen, Julie Moronuki - This is the only Haskell book I've found that is not written by a Comp Sci professor. Looks clear and easy to understand. Great introduction to functional programming. - **COST**: $59 USD

**Comment by sethburtonhall** [comment]
Frontend - Vue
---
https://michaelnthiessen.com/
Excelling free tips and [two paid courses](https://michaelnthiessen.com/#courses) 

**COST: $197 & $67**

[Vue Mastery](https://www.vuemastery.com/)
The ultimate learning resource for Vue developers. Weekly Vue.js tutorials to guide your journey to Mastery.

**COST: see [pricing plans](https://www.vuemastery.com/pricing)**

[Learn Vue](https://learnvue.co/vue-3-course)
Learn the fundamentals of Vue 3 from creating your application using Vite all the way to building and deploying your final application. We'll take you step-by-step through the process of building a complete Vue 3 app. Covering topics like the Composition API, lifecycle hooks, data, and tons more! Most importantly, you'll get access to all the code in Github repos and there is no off-screen coding so you'll be learning everything you need!

**COST: $30**

[Vue School](https://vueschool.io/courses)
Learn Vue.js and modern, cutting-edge front-end technologies from core-team members and industry experts with our premium tutorials and video courses.

**COST: see [premium plans](https://vueschool.io/plans)**

**Comment by BekahHW** [comment]
## Website templates 
- courtesy of @debrakayeelliott in slack:

- https://html5up.net/
- http://www.mashup-template.com/templates.html
- https://dev.to/rammcodes/i-created-an-opensource-portfolio-template-for-developers-1ij9
- https://wrapbootstrap.com/
- https://themeforest.net/

**Comment by BekahHW** [comment]
## Technical Writing
- https://developers.google.com/tech-writing

**Comment by BekahHW** [comment]
## React
via @LincolnFleet's post after the React meetup
https://codesandbox.io/s/github/tkshill/advanced_typescript_example2?file=/src/App2.tsx
Intro to hooks video - https://www.youtube.com/watch?v=dpw9EHDh2bM
Intro to hooks docs page - https://reactjs.org/docs/hooks-intro.html

**Reply by BekahHW** [reply]
From Meg's post in slack:

https://react.framework.dev/

**Reply by ethanfreire** [reply]
Also, check out the react community page. They have a list of free/paid courses to learn ReactJS online. https://reactjs.org/community/courses.html  & https://beta.reactjs.org/learn to learn reactJS with hooks.

**Comment by BekahHW** [comment]
## CSS & HTML

- https://webdevelopermentor.com/

**Comment by BekahHW** [comment]
- [Thinking in React](https://reactjs.org/docs/thinking-in-react.html) via @LincolnFleet from slack

**Comment by dominicduffin1** [comment]
## JavaScript

[Eloquent JavaScript](https://eloquentjavascript.net) by Marijn Haverbeke - a book about JavaScript best suited to people who have some knowledge of JavaScript or similar languages and want to level up their knowledge - can be read free online or the paperback retails at $39.95 RRP but shop around for the best price.
